virttest/utils_params.py

```diff
@@ -104,3 +105,91 @@ def copy_from_keys(self, keys):
             if self.get(key):
                 new_dict[key] = self.get(key)
         return new_dict
+
+    def get_boolean(self, key, default=False):
+        """
+        Check if a config option is set to a default affirmation or not.
+
+        :param key: config option key
+        :type key: str
+        :param bool default: whether to assume "yes" or "no" if param
+                             does not exist (defaults to False/"no").
+        :return whether option is set to 'yes'
+        :rtype: bool
+        """
+        value = self.get(key, "yes" if default else "no")
+        if value in ("yes", "on", "true"):
+            return True
+        if value in ("no", "off", "false"):
+            return False
+        raise ValueError("Cannot get boolean parameter value for %s: %s", key, value)
+
+    def get_numeric(self, key, default=0, target_type=int):
+        """
+        Get numeric value converting to integer if necessary.
+
+        :param str key: config option key
+        :param int default: default numeric value
+        :param type target_type: numeric type to return like int or float
+        :return numerical type `target_type` converted parameter value
+        :rtype: int or float
+        """
+        return target_type(self.get(key, default))
+
+    def get_list(self, key, default="", delimiter=' ', target_type=str):
+        """
+        Get a parameter value that is a character delimited list.
+
+        :param str key: parameter whose value is list
+        :param str default: default list value
+        :param str delimiter: character to split list items
+        :param type target_type: type of each item, default is string
+        :returns: empty list if if the key is not in the parameters
+        :rtype: [str]
+
+        .. note:: This an extension to the :py:func:`Params.objects` method as
+          it allows for delimiters other than white space.
+        .. seealso:: :py:func:`param_dict`
+        """
+        param_string = self.get(key, default)
+        if not param_string:
+            return []
+        else:
+            return [target_type(entry) for entry in param_string.split(delimiter)]
```

Comment:
This might produce unwanted empty entries (eg. `"foo  bar baz" => ['foo', '', 'bar', 'baz']`). Is it intended?

Reply:
We could allow more flexibility for the user even though it might result in a worse format configuration.